Convert quotas to numeric values

zakiva opened the pull request:
Parse the quotas as numeric (int or float) values instead of string. 

Depends on https://github.com/ManageIQ/manageiq-schema/pull/44

@simon3z @zeari Please review

cben commented:
@zeari how does this relate to your suffixed->value conversion quest?

zeari commented:
> @zeari how does this relate to your suffixed->value conversion quest?

Its about the same and relies on mostly the same types of conversions

zakiva commented:
@moolitayer Please review

moolitayer commented on app/models/manageiq/providers/kubernetes/container_manager/refresh_parser.rb:
```diff
@@ -869,28 +869,40 @@ def parse_resource_quota(resource_quota)
       new_result_h.values
     end
 
+    def add_numeric_quota(h, quota, key)
```
`s/key/prefix/`

zakiva replied:
done

cben commented:
@zakiva tests in #22 might help (you probably knew this)

zakiva commented:
@moolitayer `parse_quantity` method was added through #22 by @zeari with some tests (removed it from here after rebase). I updated the quotas specs and added some other example values.

moolitayer commented:
@zeari @cben review pls

moolitayer commented:
The code parsing quotas[1] is a little hard to read already and getting harder (I think I wrote it ¯\_(ツ)_/¯). 

@zakiva maybe you can add in another PR an example for the value of resource_quota.spec.<x>.to_h?

I'm sure you'll find the best way to do it

[1]
```
resource_quota.X.Y.to_h.each do |resource_name, quota|
  new_result_h[resource_name][:quota_desired] = quota
end
```

moolitayer commented:
@zakiva can you check Travis?

zakiva commented:
Updated the PR according to the schema changes, we'll probably store all values as floats (makes this PR much simpler).

Fryguy commented:
Merged the schema, so bouncing this PR.

zakiva commented:
@moolitayer this should be ready now
